### PR TITLE
[Merged by Bors] - implement signer and verifier that prefixes message

### LINKED ISSUE #3503: signer: use genesis id as domain separator for protocol messages
part of: https://github.com/spacemeshos/pm/issues/135
depends on: https://github.com/spacemeshos/go-spacemesh/issues/3505

https://github.com/spacemeshos/go-spacemesh/blob/develop/signing/signer.go

- initialize EdSigner with GenesisID (preferably pass as an option to NewEdSigner/NewEdSignerFromBuffer/NewEDVerifier)
- prefix message that is to be signed and verified with GenesisID

### PR DESCRIPTION
**Pull request by dshulyak**
closes: https://github.com/spacemeshos/go-spacemesh/issues/3503

- added option to prefix message before signing and verifying
- signing.ExtractPublicKey replaced ed25519.ExtractPublicKey, so that we can use prefixed verifier

more efficient version won't allocate new buffer when message is being prefixed, but that requires changes in spacemesh/ed25519 library

### CODE REVIEW

File: signing/signer.go
Comment (countvonzero) [inline signing/signer.go]: the only use of this func `Verify` in a test. rm?

File: cmd/node/node.go
Comment (countvonzero) [inline cmd/node/node.go]: are you waiting for the genesis confg to merge before you can initialize the signer?

**Review by countvonzero — APPROVED** [review]

File: cmd/node/node.go
Comment (dshulyak) [reply cmd/node/node.go]: yes, i added a placeholder genesis id initially but then realized that it is better to rebase after config is merged

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**
